src/test/java/no/fintlabs/operator/PGUserSecretDependentResourceTest.java

```diff
@@ -0,0 +1,106 @@
+package no.fintlabs.operator;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.Matcher;
+import no.fintlabs.FlaisWorkflow;
+import no.fintlabs.OperatorProperties;
+import no.fintlabs.aiven.AivenService;
+import no.fintlabs.pg.PgService;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Base64;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class PGUserSecretDependentResourceTest {
+    @Test
+    public void testDesired() {
+        PGUserCRD primary = new PGUserCRD();
+        Context<PGUserCRD> context = mock(Context.class);
+
+        PGUserSecretDependentResource pgUserSecretDependentResource = new PGUserSecretDependentResource(
+                mock(FlaisWorkflow.class),
+                mock(KubernetesClient.class),
+                mock(PGUserDependentResource.class),
+                mock(OperatorProperties.class)
+        );
+
+        PGUser pgUser = new PGUser();
+        pgUser.setUsername("testUser");
+        pgUser.setPassword("testPassword");
+        pgUser.setDatabase("testDatabase");
+
+        when(context.getSecondaryResource(PGUser.class)).thenReturn(Optional.of(pgUser));
+
+        Secret secret = pgUserSecretDependentResource.desired(primary, context);
+
+        byte[] decodedUsername = Base64.getDecoder().decode(secret.getData().get("fint.database.username"));
+        String decodedUsernameString = new String(decodedUsername);
+
+        assertEquals(pgUser.getUsername(), decodedUsernameString);
+
+        when(context.getSecondaryResource(PGUser.class)).thenReturn(Optional.empty());
+
+        assertThrows(RuntimeException.class, () -> {
+            pgUserSecretDependentResource.desired(primary, context);
+        });
```

Comment:
Ikke sjekk for to potensielle scenarioer i en test. Splitt dette ut i en separat test